app/src/main/java/me/saket/dank/widgets/IndentedLayout.java

```diff
@@ -54,31 +94,57 @@ public IndentedLayout(Context context, @Nullable AttributeSet attrs) {
   protected void onLayout(boolean changed, int l, int t, int r, int b) {
     super.onLayout(changed, l, t, r, b);
 
-    indentationLinePath.reset();
     for (int i = 0; i < indentationDepth; i++) {
       float lineStartX = spacePerDepthPx * (i + 1) + indentationLinePaint.getStrokeWidth();
-      indentationLinePath.moveTo(lineStartX, 0);
-      indentationLinePath.lineTo(lineStartX, getHeight());
+
+      this.trees.get(i).path.moveTo(lineStartX, 0);
```

Comment:
Could create local variable for `trees.get(i).path` to reduce code duplication.

Also similar to how it was done before, but this time for each path, you should call `reset` method on them. This will make sure that there will be always one line drawn. Otherwise this code will add new line to the path on top of previous each time `onLayout` is called.

Lastly `this.` prefix is not necessary. It's my style preference but I would suggest to use it only where needed. This suggestion applies to similar occurrences in this file.